src/engine/strat_engine/engine.rs

```diff
@@ -122,6 +141,42 @@ impl Engine for StratEngine {
     fn pools(&self) -> Vec<&Pool> {
         self.pools.into_iter().map(|x| x as &Pool).collect()
     }
+
+    fn get_eventable(&mut self) -> EngineResult<Option<Box<Eventable>>> {
+        Ok(Some(Box::new(DM::new()?)))
+    }
+
+    fn evented(&mut self) -> EngineResult<()> {
+        let device_list = DM::new()?.list_devices()?;
+
+        for &(ref dm_name, _, new_event_nr) in &device_list {
+            if let Some(new_event_nr) = new_event_nr {
```

Comment:
I think you can drop both these if's. Instead,

```if new_event_nr > self.watched_devs.get_mut(dm_name) {```

should work out correctly.

Reply:
I think there was a theoretical concern about event_nr wrapping?

Reply:
Actually, I think there may be a bug in your current formulation. I think that if ```watched_devs``` lacks an entry for ```dm_name``` but there was a corresponding ```event_nr``` for the name in the device list then you had better take the action. But in your current code, you don't.

Reply:
If watched_devs lacks an entry, then an event happened on a DM device we don't care about. DM::list_devices() returns all DM devices on the system. If e.g. an LVM configured DM device has an event we will still end up in this function and should take no action.

Reply:
I don't think that's quite right.

Last time we called ```evented()```, a particular device might not have existed and ```watched_dev_last_event_nrs``` would not contain a key for that device. But this time we call ```evented()``` the device does exist, so it is contained in ```list_devices()``` result but not in ```watched_dev_last_event_nrs```. I think we need to do something in that case.